### ISSUE TITLE
Wrapping Vectors in Options seems unnecessary

### ISSUE THREAD

**Issue by Manghi** [conwayste/conwayste#106]
Consider the `Update` variant of the `Packet` enum.

```Rust
pub enum Packet {
...
    Update {
        chats:           Option<Vec<BroadcastChatMessage>>,
        game_updates:    Option<Vec<GameUpdate>>,
        universe_update: UniUpdateType,
    },
...
}
```

The first two fields, `chats` and `game_updates`, are lists wrapped by the `Option` enumerated type. 

I'm proposing we get rid of the `Option`. There really isn't any need for the `Option`-wrapping, because in the `None` case we could simply have an empty list. I think the space saving benefits of using the Option doesn't gain us much, and instead increases the complexity of the code needed to access the inner item. 

The definition would be rewritten as:

```Rust
pub enum Packet {
...
    Update {
        chats:           Vec<BroadcastChatMessage>,
        game_updates:    Vec<GameUpdate>,
        universe_update: UniUpdateType,
    },
...
}
```

This is specifically an issue in the `netwayste` code (and was written by me, ha!), and I think it was over-implemented on the first go. 

@AaronM04 
Any objections to cleaning this up?

**Comment by AaronM04** [comment]
No objections at all. Please proceed! 😁

On April 19, 2020 3:22:47 PM PDT, Manghi <notifications@github.com> wrote:
>Consider the `Update` variant of the `Packet` enum.
>
>```Rust
>pub enum Packet {
>...
>    Update {
>        chats:           Option<Vec<BroadcastChatMessage>>,
>        game_updates:    Option<Vec<GameUpdate>>,
>        universe_update: UniUpdateType,
>    },
>...
>}
>```
>
>The first two fields, `chats` and `game_updates`, are lists wrapped by
>the `Option` enumerated type. 
>
>I'm proposing we get rid of the `Option`. There really isn't any need
>for the `Option`-wrapping, because in the `None` case we could simply
>have an empty list. I think the space saving benefits of using the
>Option doesn't gain us much, and instead increases the complexity of
>the code needed to access the inner item. 
>
>The definition would be rewritten as:
>
>```Rust
>pub enum Packet {
>...
>    Update {
>        chats:           Vec<BroadcastChatMessage>,
>        game_updates:    Vec<GameUpdate>,
>        universe_update: UniUpdateType,
>    },
>...
>}
>```
>
>This is specifically an issue in the `netwayste` code (and was written
>by me, ha!), and I think it was over-implemented on the first go. 
>
>@AaronM04 
>Any objections to cleaning this up?
>
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/conwayste/conwayste/issues/106

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by Manghi** [comment]
Since this is such a small change and somewhat similar to #105, it has been implemented and is under  review via #107.

**Comment by Manghi** [comment]
This has been resolved in #107.
